src/TextParser.php

```diff
@@ -72,7 +72,7 @@ protected function getTemplates($text, $findMatchingTemplate)
 
         $templates = [];
         foreach ($this->directoryIterator as $fileInfo) {
-            $templates[$fileInfo->getPathname()] = file_get_contents($fileInfo->getPathname());
+            if(!is_dir($fileInfo->getPathname()))$templates[$fileInfo->getPathname()] = file_get_contents($fileInfo->getPathname());
```

Comment:
Looks fine, I would just avoid having the condition and the array filling on the same line (makes it bit confusing)

I might also want to expand this skip everything that's not readable, so not only directories in this case, maybe something like:

```        
       foreach ($this->directoryIterator as $fileInfo) {
            if (!is_file($fileInfo->getPathname())) {
                continue;
            }
            $templates[$fileInfo->getPathname()] = file_get_contents($fileInfo->getPathname());
        }
```